devchat.go

```diff
@@ -437,12 +438,52 @@ func (u *user) pickUsername(possibleName string) {
 		possibleName = cleanName(possibleName)
 	}
 	u.name = possibleName
-	u.changeColor(*colorArr[rand.Intn(len(colorArr))]) // also sets prompt
+	var colorIndex = rand.Intn(len(colorArr))
+	u.changeColor(colorNames[colorIndex], *colorArr[colorIndex]) // also sets prompt
 }
 
-func (u *user) changeColor(color color.Color) {
+func (u *user) changeColor(colorName string, color color.Color) {
 	u.name = color.Sprint(stripansi.Strip(u.name))
-	u.color = color
+	u.color = colorName
+	u.term.SetPrompt(u.name + ": ")
+	allUsersMutex.Lock()
+	allUsers[u.id] = u.name
+	allUsersMutex.Unlock()
+	saveBansAndUsers()
+}
+
+func (u *user) getColor() *color.Color {
+	for i := 0; i < len(colorNames); i++ {
+		if colorNames[i] == u.color {
+			return colorArr[i]
+		}
+	}
+	return colorArr[0]
+}
+
+func (u *user) rainbowColor() {
+	var rainbow = [...]color.Color{
+		*green,
```

Comment:
(about line 466) If I remember correctly, these (green, yellow, etc.) are already pointers to a color, so it might be a good idea to not take a pointer of them.

Reply:
Good point. Removed the references!

Reply:
Thanks!